tensorbay/opendataset/SegTrack2/loader.py

```diff
@@ -0,0 +1,156 @@
+#!/usr/bin/env python3
+#
+# Copytright 2021 Graviti. Licensed under MIT License.
+#
+# pylint: disable=invalid-name
+# pylint: disable=missing-module-docstring
+
+import os
+from typing import Callable
+
+import numpy as np
+
+from ...dataset import Data, Dataset
+from ...label import InstanceMask
+from .._utility import glob
+
+try:
+    from PIL import Image
+except ModuleNotFoundError:
+    from .._utility.mocker import Image
+
+DATASET_NAME = "SegTrack2"
+
+_SEGMENTS_INFO = {
+    1: ("bird_of_paradise", "birdfall", "frog", "girl", "monkey", "parachute", "soldier", "worm"),
+    2: ("bmx", "cheetah", "drift", "hummingbird", "monkeydog"),
+    6: ("penguin",),
+}
+_FILENAME_REFORMATTERS = (
+    lambda filename: filename,
+    lambda filename: f"{os.path.splitext(filename)[0]}.png",
+)
+_MASK_GETTER = Callable[[str, str, str, int, Callable[[str], str]], InstanceMask]
+
+
+def SegTrack2(path: str) -> Dataset:
+    """Dataloader of the `SegTrack2`_ dataset.
+
+    .. _SegTrack: https://web.engr.oregonstate.edu/~lif/SegTrack2/dataset.html
+
+    The file structure of SegTrack looks like::
+
+        <path>
+            GroundTruth/
+                bird_of_paradise/
+                    bird_of_paradise_00000.png
+                    ...
+                bmx/
+                    1/
+                        bmx_06668.png
+                        ...
+                    2/
+                        bmx_06668.png
+                        ...
+                ...
+            JPEGImages/
+                bird_of_paradise/
+                    bird_of_paradise_00000.png
+                    ...
+                ...
+
+    Arguments:
+        path: The root directory of the dataset.
+
+    Returns:
+        Loaded :class:`~tensorbay.dataset.dataset.Dataset` instance.
+
+    """
+    root_path = os.path.join(os.path.abspath(os.path.expanduser(path)))
+    image_dir = os.path.join(root_path, "JPEGImages")
+    original_mask_dir = os.path.join(root_path, "GroundTruth")
+    mask_dir = os.path.join(root_path, "Masks")
+    dataset = Dataset(DATASET_NAME)
+    dataset.notes.is_continuous = True
+    dataset.load_catalog(os.path.join(os.path.dirname(__file__), "catalog.json"))
+    _load_segments(image_dir, original_mask_dir, mask_dir, dataset)
+    return dataset
+
+
+def _load_segments(image_dir: str, original_mask_dir: str, mask_dir: str, dataset: Dataset) -> None:
```

Comment:
it will be better to make function const, which means the function will not change the input arguments.